api/v1alpha1/shipwrightbuild_types.go

```diff
@@ -51,3 +55,20 @@ type ShipwrightBuildList struct {
 func init() {
 	SchemeBuilder.Register(&ShipwrightBuild{}, &ShipwrightBuildList{})
 }
+
+// SetCondition updates a list of conditions with the provided condition
+func (sbs *ShipwrightBuildStatus) SetCondition(condition *buildcorev1alpha1.Condition) {
+	var idx = -1
+	for i, c := range sbs.Conditions {
+		if c.Type == condition.Type {
+			idx = i
+			break
+		}
+	}
+
+	if idx != -1 {
+		sbs.Conditions[idx] = *condition
+	} else {
+		sbs.Conditions = append(sbs.Conditions, *condition)
+	}
+}
```

Comment:
```suggestion
func (sbs *ShipwrightBuildStatus) SetCondition(condition *buildcorev1alpha1.Condition) {
	for i, c := range sbs.Conditions {
		if c.Type == condition.Type {
			sbs.Conditions[i] = *condition
			return
		}
	}
	sbs.Conditions = append(sbs.Conditions, *condition)
}
```

Reply:
I am disappointed that Kubernetes does not have methods that they provide for dealing with performing CRUD operations on Conditions.

Reply:
Openshift are doing it here https://github.com/openshift/custom-resource-status/blob/master/conditions/v1/conditions.go
but I used the same approch used in the shipwright/build projet https://github.com/shipwright-io/build/blob/main/pkg/apis/build/v1alpha1/buildrun_types.go#L220